### DISCUSSION TITLE
How do I view Helm NOTES.txt output for flux HelmReleases?

### DISCUSSION THREAD

**Discussion by brsolomon-deloitte** [fluxcd/flux2#4059] [Q&A] [answered]
How do I view the rendered Helm chart [`NOTES.txt`](https://helm.sh/docs/chart_template_guide/notes_files/) for a flux `HelmRelease` using `kubectl` or `flux` CLI?

I've tried:

```bash
kubectl describe hr -n <namespace> nexus-repository-manager
flux logs --kind HelmRelease -n <namespace>
```

**Comment by stefanprodan** [comment, marked as answer]
With `helm -n <HR NAMESPACE> get notes <HR NAME>`, docs here https://helm.sh/docs/helm/helm_get_notes/

**Comment by brsolomon-deloitte** [comment]
> With `helm -n <HR NAMESPACE> get notes <HR NAME>`, docs here https://helm.sh/docs/helm/helm_get_notes/

That doesn't seem to be working, possibly due to https://github.com/helm/helm/issues/10844

```shell
$ helm list -a -A | egrep 'NAME|nexus-repo'
NAME                            NAMESPACE                   REVISION    UPDATED                                 STATUS          CHART                                           APP VERSION
nexus-repository-manager        nexus-repository-manager    1           2023-07-10 12:08:25.480847057 +0000 UTC deployed        nexus-repository-manager-<redacted>             3.47.1

$ helm get notes -n nexus-repository-manager nexus-repository-manager
Error: release: not found

$ kubectl get hr -A | egrep 'NAME|nexus-repo'
NAMESPACE   NAME                       AGE    READY   STATUS
<redacted>     nexus-repository-manager   125m   True    Release reconciliation succeeded
```

**Reply by hiddeco** [reply]
From the top of my head, the namespace listed by `helm list` is the release namespace (and not the namespace the Secret is stored in).

My theory is that you likely have a `targetNamespace` set without a `storageNamespace` configured, this means that for `helm get` you would need to run `helm get -n <redacted>` to make Helm reach out to the Secret in the appropriate namespace.
